app/models/anonymous_user.rb

```diff
@@ -122,4 +122,21 @@ def school_type=(type)
     raise AnonymousUserIsImmutableError, "Cannot set school type on the AnonymousUser."
   end
 
+  User.school_locations.each do |location, value|
+    define_method "#{location}?" do
+      User::DEFAULT_SCHOOL_LOCATION.to_s == location
+    end
+
+    define_method "#{location}!" do
+      raise AnonymousUserIsImmutableError, "Cannot set school location on the AnonymousUser."
+    end
+  end
+
+  def school_location
+    User::DEFAULT_SCHOOL_LOCATION.to_s
+  end
+
+  def school_location=(type)
+    raise AnonymousUserIsImmutableError, "Cannot set school location on the AnonymousUser."
+  end
 end
```

Comment:
will school locations always be valid ruby method names?  For instance this would fail if they had a `'` or something in the name

Reply:
They are an enum so they have to be valid... The 3 we have currently are:
```
unknown_school_location
domestic_school
foreign_school
```